src/pc/1.8/chunk.js

```diff
@@ -139,44 +140,47 @@ class Chunk {
   }
 
   dump (bitMap = 0xFFFF, skyLightSent = true) {
-    const SECTION_SIZE = Section.sectionSize(skyLightSent)
+    const SECTION_SIZE = Section.sectionSize(this.skyLightSent && skyLightSent)
 
     const { chunkIncluded, chunkCount } = parseBitMap(bitMap)
     const bufferLength = chunkCount * SECTION_SIZE + BIOME_SIZE
     const buffer = Buffer.alloc(bufferLength)
     let offset = 0
     let offsetLight = w * l * sectionCount * chunkCount * 2
-    let offsetSkyLight = w * l * sectionCount * chunkCount / 2 * 5
+    let offsetSkyLight = (this.skyLightSent && skyLightSent) ? w * l * sectionCount * chunkCount / 2 * 5 : undefined
     for (let i = 0; i < sectionCount; i++) {
       if (chunkIncluded[i]) {
         offset += this.sections[i].dump().copy(buffer, offset, 0, w * l * sh * 2)
         offsetLight += this.sections[i].dump().copy(buffer, offsetLight, w * l * sh * 2, w * l * sh * 2 + w * l * sh / 2)
-        offsetSkyLight += this.sections[i].dump().copy(buffer, offsetSkyLight, w * l * sh / 2 * 5, w * l * sh / 2 * 5 + w * l * sh / 2)
+        if (this.skyLightSent && skyLightSent) offsetSkyLight += this.sections[i].dump().copy(buffer, offsetSkyLight, w * l * sh / 2 * 5, w * l * sh / 2 * 5 + w * l * sh / 2)
       }
     }
-    this.biome.copy(buffer, w * l * sectionCount * chunkCount * 3)
+    this.biome.copy(buffer, w * l * sectionCount * chunkCount * ((this.skyLightSent && skyLightSent) ? 3 : 5 / 2))
     return buffer
   }
 
   load (data, bitMap = 0xFFFF, skyLightSent = true) {
     if (!Buffer.isBuffer(data)) { throw (new Error('Data must be a buffer')) }
 
+    this.skyLightSent = skyLightSent
+
     const SECTION_SIZE = Section.sectionSize(skyLightSent)
 
     const { chunkIncluded, chunkCount } = parseBitMap(bitMap)
+    console.log('cc: ' + chunkCount)
     let offset = 0
     let offsetLight = w * l * sectionCount * chunkCount * 2
-    let offsetSkyLight = w * l * sectionCount * chunkCount / 2 * 5
+    let offsetSkyLight = (this.skyLightSent) ? w * l * sectionCount * chunkCount / 2 * 5 : undefined
```

Comment:
why undefined ? that doesn't make sense

Reply:
Because at line 127 I reference offsetSkyLight when this.skyLightSent is true. So the assigned value of offsetSkyLight will only be undefined when this.skyLightSent is actually false, and that state won't change no matter what til line 179.
And because of that I choose undefined(rather than null which is still a obj) to not unnecessary calcute even once a offset for something I will never use. At first it looked like this to not even declare the variable to begin with 
`if (this.skyLightSent) let offsetSkyLight = (this.skyLightSent) ? w * l * sectionCount * chunkCount / 2 * 5`
and therefor be even faster. But since the variable would still needed to be referenced at line 179 even when only the condition is true I couldn't get around the declaration of offsetSkyLight so I thought atleast don't waste calculation time on an offset for a variable from which we already know that it won't get used but still must be declared to not cause errors.